presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/CheckNoIPAddressType.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.type.ArrayType;
+import com.facebook.presto.common.type.MapType;
+import com.facebook.presto.common.type.RowType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.expressions.DefaultRowExpressionTraversalVisitor;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.WarningCollector;
+import com.facebook.presto.spi.plan.AggregationNode;
+import com.facebook.presto.spi.plan.FilterNode;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.ProjectNode;
+import com.facebook.presto.spi.plan.ValuesNode;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.InputReferenceExpression;
+import com.facebook.presto.spi.relation.IntermediateFormExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.SimplePlanVisitor;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+
+import java.util.List;
+
+import static com.facebook.presto.type.IpAddressType.IPADDRESS;
+import static com.google.common.base.Preconditions.checkState;
+
+public class CheckNoIPAddressType
```

Comment:
Instead of adding one more plan visitor, I wonder if adding to existing plan visitor is a better idea.

Reply:
@mohsaka  : +1

Please add the changes along with TimestampWithTimezone changes https://github.com/prestodb/presto/pull/23200 and https://github.com/prestodb/presto/pull/23374

Reply:
Will do

Reply:
@feilong-liu @aditi-pandit 

So I made a change to the error message string and hasType to support IPAddress type.

However I have a few questions.

1. Should we support/not support Timestamp With Timezone and IPAddress independently? Say for example Timestamp with timezone order by is fixed before IPAddress, then we can disable the types independently? If so I believe that we would have to remove the static final from the error message and in turn remove the static from the nested classes. Since I think the best way to do this would be to set the error message in `hasType` and exit on first occurrence of either type. 

We can also make it more complex by having `hasType` do a complete recursion or at least until all types are hit to make a more complex error message with either or both types included in it. 

If not. 
2. Currently I'm only using the `disableTimeStampWithTimeZoneForNative` config. As an or/and with both configs don't make sense. I was planning on renaming it to something like disablePhysicalMappedTypes. However it seems like this config is used, and I do not want to break deployment configs.
https://github.com/prestodb/presto/pull/23200#discussion_r1676659424

Reply:
@mohsaka : We should enable/disable each independently.

I'm not entirely familiar with Java but is it possible to abstract a common base class to do PlanChecking that forwards hasType() to its child class. We could have 2 child classes for either type.

Reply:
@mohsaka you can pass the flag of disabling timestamp and iptype to the checker, and in the plan checker do the check accordingly based on the flag

Reply:
@feilong-liu Agreed. I think I'm doing something similar to that. Let me push the changes once complete and please take a look. Thanks!

Reply:
@feilong-liu @aditi-pandit Pushed the changes. Please take a look. 

Decided not to set the member variable string as I did not know if there are multiple callers on different threads. Could cause an issue where a string from one execution's string could appear as an error on another if run at the same time due to static.

Also currently made is so that we do not recurse through the entire plan as it was before. We could still do this and make a more informative string. But I think its unnecessary. 
Ex if there is a Timestamp with timezone type and IPAddress type then it would show both in one error message.